### ISSUE TITLE
 #[serde(alias = "somename")] stopped working after 0.6

### ISSUE THREAD

**Issue by VegarRingdalAibel** [RReverser/serde-wasm-bindgen#63]
After when going from 0.5 to 0.6 it looks like alias is broken

Using:
```
wasm-bindgen = "0.2.91"
serde = { version = "1.0.196", features = ["derive"] }
serde_json = "1.0.113"
serde-wasm-bindgen = "0.6.3" //0.5 works... fails if i go higher
console_error_panic_hook="0.1.7"
```

```rs
#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum FilterAction {
    Append,
    Remove,
    Keep,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub enum FilterType {
    Standard,
    Boundingbox,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct FilterInput {
    #[serde(alias = "filtertype")]
    pub filter_action: FilterAction,
    #[serde(alias = "type")]
    pub filter_type: FilterType,
    ...

```

This is my input
![image](https://github.com/RReverser/serde-wasm-bindgen/assets/94840334/8a3e091d-f281-4e69-953f-0c1f4f9db22f)

get error:
```
panicked at src\lib.rs:228:58:
called `Result::unwrap()` on an `Err` value: Error(JsValue(Error: unknown variant `Append`, expected `Standard` or `Boundingbox`
Error: unknown variant `Append`, expected `Standard` or `Boundingbox`
```

Its like its setting values from json `filtertype` to rust `filter_type` when it should have used json `type`.

Is it a bug or have I just been lucky it been working ?

**Comment by RReverser** [comment]
Hm... the last change that transitioned from 0.5 to 0.6 was #40, but I don't see why it would cause this particular breakage.

**Comment by RReverser** [comment]
To be clear, does it break on transition to any 0.6 or only 0.6.3? Did you try pinning exact versions so that Cargo doesn't fetch the latest 0.6.x?

**Comment by VegarRingdalAibel** [comment]
I'll do a new try later this week, so I know better when it breaks. I can try and create a simple sample repo with the bug. 👍

**Comment by VegarRingdalAibel** [comment]
@RReverser 
Made a small sample containing 0.5.0(wasm1) and 0.6.0 (wasm2)

https://github.com/vegarringdal/wasm_test

![image](https://github.com/RReverser/serde-wasm-bindgen/assets/94840334/e9a40010-e368-4b79-bffa-5a9fe7933f3a)

**Comment by RReverser** [comment]
You can see from the Cargo.lock that wasm2 is not 0.6.0 but 0.6.3: https://github.com/vegarringdal/wasm_test/blob/9238388d3217b95fd3cd86bc4a6598c05af72c63/wasm2/Cargo.lock#L87-L89

That's what I suspected and why I mentioned pinning exact version to narrow it down further. I believe you can do that by specifying version as `=0.6.0` instead of `0.6.0`.

**Comment by VegarRingdalAibel** [comment]
Ill have a new look, though it would use the version I had here
https://github.com/vegarringdal/wasm_test/blob/9238388d3217b95fd3cd86bc4a6598c05af72c63/wasm2/Cargo.toml#L14

**Comment by RReverser** [comment]
Yeah, no, Cargo auto-updates the version to whatever is compatible according to semver. It's usually the right choice, but not very helpful when debugging an issue.

**Comment by VegarRingdalAibel** [comment]
I have clearly missed this part 😂

**Comment by VegarRingdalAibel** [comment]
Error is after version 0.6.2, update repo

**Comment by RReverser** [comment]
> Error is after version 0.6.2, update repo

What do you mean by "update repo"? Looks like you didn't push any new changes?

**Comment by RReverser** [comment]
That said, if it's 0.6.2+ rather than **after** 0.6.2 (0.6.3+), then I suspect it might be due to this change https://github.com/RReverser/serde-wasm-bindgen/commit/ce7669e1d175d999a733497e87152f55808ee8f2

**Comment by VegarRingdalAibel** [comment]
HI
I pushed this change :-)

https://github.com/vegarringdal/wasm_test/commit/0b2fa8a8e28f78150b2a2737fe3c6785164d2241

![image](https://github.com/RReverser/serde-wasm-bindgen/assets/94840334/1cbe88e8-553a-430c-ae10-b597fde57b4f)


> That said, if it's 0.6.2+ rather than after 0.6.2 (0.6.3+), then I suspect it might be due to this change > > https://github.com/RReverser/serde-wasm-bindgen/commit/ce7669e1d175d999a733497e87152f55808ee8f2

Not sure if Serde supports field indices in combination with aliases.


I dont know :-)

**Comment by RReverser** [comment]
Fixed in 0.6.4. Thanks for the report!

**Comment by VegarRingdalAibel** [comment]
Thank you.
